crates/rpc/src/v02/method/get_transaction_receipt.rs

```diff
@@ -665,81 +648,10 @@ mod tests {
-            transaction_hash: transaction_hash_bytes!(b"txn 6"),
-        };
-
-        let result = get_transaction_receipt(context, input).await.unwrap();
-        use types::*;
-        assert_eq!(
-            result,
-            MaybePendingTransactionReceipt::Normal(TransactionReceipt::Invoke(
-                InvokeTransactionReceipt {
-                    common: CommonTransactionReceiptProperties {
-                        transaction_hash: transaction_hash_bytes!(b"txn 6"),
-                        actual_fee: Fee::ZERO,
-                        status: TransactionStatus::AcceptedOnL2,
-                        block_hash: block_hash_bytes!(b"latest"),
-                        block_number: BlockNumber::new_or_panic(2),
-                        messages_sent: vec![MessageToL1 {
-                            from_address: None, // RPC v0.2 does not have this field
-                            to_address: EthereumAddress(H160::zero()),
-                            payload: vec![
-                                l2_to_l1_message_payload_elem!("0x1"),
-                                l2_to_l1_message_payload_elem!("0x2"),
-                                l2_to_l1_message_payload_elem!("0x3"),
-                            ],
-                        }],
-                        events: vec![],
-                    }
-                }
-            ))
-        )
-    }
-
-    #[tokio::test]
-    async fn success_v03() {
-        let context = RpcContext::for_tests().with_version("v0.3");
-        let input = GetTransactionReceiptInput {
-            transaction_hash: transaction_hash_bytes!(b"txn 6"),
-        };
-
-        let result = get_transaction_receipt(context, input).await.unwrap();
-        use types::*;
-        assert_eq!(
-            result,
-            MaybePendingTransactionReceipt::Normal(TransactionReceipt::Invoke(
-                InvokeTransactionReceipt {
-                    common: CommonTransactionReceiptProperties {
-                        transaction_hash: transaction_hash_bytes!(b"txn 6"),
-                        actual_fee: Fee::ZERO,
-                        status: TransactionStatus::AcceptedOnL2,
-                        block_hash: block_hash_bytes!(b"latest"),
-                        block_number: BlockNumber::new_or_panic(2),
-                        messages_sent: vec![MessageToL1 {
-                            from_address: Some(contract_address!("0xcafebabe")),
-                            to_address: EthereumAddress(H160::zero()),
-                            payload: vec![
-                                l2_to_l1_message_payload_elem!("0x1"),
-                                l2_to_l1_message_payload_elem!("0x2"),
-                                l2_to_l1_message_payload_elem!("0x3"),
-                            ],
-                        }],
-                        events: vec![],
-                    }
-                }
-            ))
-        )
-    }
-
```

Comment:
Can you confirm these aren't relevant anymore?

Reply:
This test (and the other deletions) are remnants from when we served RPC spec v0.2 and v0.3. There was a breaking change from v0.2 to v0.3 which impacted receipt serialization. However, unlike our other RPC methods (which have separate impls for methods that change), this was handled by injecting the RPC version explicitly and relying on the same implementation tweaked based on the version.

We now serve v0.3 and v0.4 and no longer need to inject the version (all versions are the same), so this is removing something that was no longer needed.

A better way to handle these differences (often its just different serialization) is to have a common impl, and then the specific method can just wrap the output with its serialization type. We're slowly getting to something like this overtime.